DateUtility.js

```diff
@@ -0,0 +1,239 @@
+/*************************************************************************
+* Name       : Huan Doan
+* Course     : JavaScript 300
+* Description: Revealing Module Pattern module that provides
+*              select Date functions.
+*
+* Anytime a method is called that needs to access your internally stored
+* date, you'll want to make sure you've instantiated it first. Your
+* module should not start out with this date being set to anything yet.
+* If it hasn't yet been instantiated, instantiate it then (this is a good
+* place to use a "private" method.) If it has already been instantiated,
+* don't re-instantiate it (with the exception of setDate which should
+* instantiate your date again based on how it's called).
+*
+* In your module file, assign the return value of your module to a global
+* variable.
+*************************************************************************/
+var customDate = (function wrapperIIFE() {
+
+  // private variable - the internally stored date
+  var datePrivate;
+
+  function assignDate(value) {
+    if (!datePrivate) {
+      datePrivate = new Date();
+    }
+    datePrivate.setTime(value);
+  }
+
+  function getFormattedDate() {
+    var formattedDate = '';
+		
+    if (!datePrivate) {
+      datePrivate = new Date();
+    }
+
+    formattedDate = getMonthNamePublic() + ' ' +
+      datePrivate.getDate() + ', ' + datePrivate.getFullYear();
+
+    return formattedDate;
+  }
+
+  function getDateMsec() {
+    if (!datePrivate) {
+      datePrivate = new Date();
+    }
+	
+    return datePrivate.getTime();
+  }
+
+  function isObjectFormat(value) {
+    return ((value !== null) &&
+            (typeof value === 'object'));
+  }
+
+  /***************************************************************************
+  * Allows the internally stored date to be returned to the user in one of
+  * two formats: as milliseconds after "epoch" or as a date string in
+  * the following format: "[Month] [date], [year]". This method should take
+  * an options parameter that specifies a format property, with a value of
+  * "milliseconds" or "formatted". If the options argument is ommited,
+  * default to milliseconds.
+  ***************************************************************************/
+  function getDatePublic(formatOfDate) {
+    if ((arguments.length === 1) &&
+        (isObjectFormat(formatOfDate)) &&
+        (formatOfDate['format'] === 'formatted')) {
+      return getFormattedDate();
+    } else {
+      return getDateMsec();
+    }
+  }
+
+  /***************************************************************************
+  * Returns the name of the day to be returned for the internally stored
+  * date - e.g. "Tuesday", or "Wednesday". This method takes no parameters.
+  ***************************************************************************/
+  function getDayNamePublic() {
+    var dayName = '';
+
+    if (!datePrivate) {
+      datePrivate = new Date();
+    }
+	
+    switch(datePrivate.getDay()) {
```

Comment:
While this works, a simpler approach would just be to use an array of strings and index into them, using the same value you have cases for.